.circleci/config.yml

```diff
@@ -1,63 +1,91 @@
-              python setup.py sdist
-              pip install dist/`ls dist/ | grep .tar.gz`
-      - run:
-          name: Install the full development requirements
-          working_directory: ~/project/
-          command: pip install --progress-bar off -r requirements.txt
-      - run:
-          name: Show installed Python packages
-          command: pip list
+          name: Create custom requirements
+          command: grep << parameters.command-name >> requirements.txt > << parameters.command-name >>_requirements.txt
+      - python/install-packages:
+          pkg-manager: pip
+          pip-dependency-file: << parameters.command-name >>_requirements.txt
+          cache-version: << parameters.command-name >>-v1
       - run:
-          name: Lint
           working_directory: ~/
-          # Avoid being able to import wikipron by relative import.
-          # Test code by importing the *installed* wikipron in site-packages.
-          command: flake8 project/setup.py project/wikipron project/tests
+          command: << parameters.command-run >>
+
+  build-python:
+    parameters:
+      python-version:
+        type: string
+    docker:
+      - image: cimg/python:<< parameters.python-version >>
+        #auth:
+        #  username: $DOCKERHUB_USERNAME
+        #  password: $DOCKERHUB_PASSWORD
+    steps:
+      - checkout
       - run:
-          name: Check styling consistency with black
-          working_directory: ~/
-          # Avoid being able to import wikipron by relative import.
-          # Test code by importing the *installed* wikipron in site-packages.
-          command: black --line-length=79 --check project/setup.py project/wikipron project/tests project/data
+          name: Create empty pyenv file
+          command: |
+              mkdir -p /home/circleci/.pyenv/versions/
+              touch /home/circleci/.pyenv/version
+      - python/install-packages:
+          pkg-manager: pip
+          pre-install-steps:
+              - run:
+                  name: Build source distribution and install package from it
+                  command: |
+                      pip install --progress-bar off --upgrade pip setuptools
+                      python setup.py sdist
+                      pip install dist/*.tar.gz
+      - run:
+          name: Show installed Python packages
+          command: pip3 list -v
       - run:
           name: Run python tests
           working_directory: ~/
           # Avoid being able to import wikipron by relative import.
           # Test code by importing the *installed* wikipron in site-packages.
-          command: pytest -vv project/tests
-  build-python-3.7:
-    <<: *template
-    docker:
-      - image: python:3.7
-  build-python-3.8:
-    <<: *template
-    docker:
-      - image: python:3.8
-  build-python-3.9:
-    <<: *template
-    docker:
-      - image: python:3.9
+          command: |
+              sudo chown circleci:circleci /home
```

Comment:
Why do we need to run `chown` on `/home` here? A comment just above would help.

Reply:
I noticed that the [first test are failing](https://app.circleci.com/pipelines/github/kylebgorman/wikipron/560/workflows/997720a3-3515-4d7a-a01c-20604a8bbc5b/jobs/1642/tests) because they are trying to create a directory in `/home` which is owned by `root`.
I chose a quick and dirty fix to make tests pass; of course the correct fix is to move the test directory to either a unique temporary folder via `tempfle` stdlib or a fixed directory in a user owned path like `~/tsv`

Reply:
Oh yes, we do have tests with file I/O like that. Your quick fix here seems reasonable for now.

Note to self: I did [mention](https://github.com/kylebgorman/wikipron/pull/226) using `tempfile` when the relevant tests were being checked in. This issue will be fixed when we get to refactoring the relevant code later.